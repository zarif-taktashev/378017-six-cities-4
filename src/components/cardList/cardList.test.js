import React from "react";
import renderer from "react-test-renderer";
import CardList from "./cardList.jsx";

const offers = [
  {
    img: `img/apartment-01.jpg`,
    premium: true,
    costs: 120,
    banner: `1 Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    rate: 80,
    coordinates: [52.3909553943508, 4.85309666406198]
  },
  {
    img: `img/apartment-02.jpg`,
    premium: false,
    costs: 50,
    banner: `2 Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    rate: 70,
    coordinates: [52.369553943508, 4.85309666406198]
  },
  {
    img: `img/apartment-03.jpg`,
    premium: true,
    costs: 80,
    banner: `3 Beautiful & luxurious apartment at great location`,
    type: `Private room`,
    rate: 20,
    coordinates: [52.3909553943508, 4.929309666406198]
  },
  {
    img: `img/apartment-01.jpg`,
    premium: false,
    costs: 10,
    banner: `4 Beautiful & luxurious apartment at great location`,
    type: `Private room`,
    rate: 50,
    coordinates: [52.3809553943508, 4.939309666406198]
  }
];

describe(`main-test`, () => {
  it(`Main unit test`, () => {
    const tree = renderer
      .create(<CardList
        offers={offers}
        onHandleChosenCard={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
