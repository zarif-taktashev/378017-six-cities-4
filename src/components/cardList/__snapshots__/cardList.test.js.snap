// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main-test Main unit test 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            813
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button  button"
          href="/login"
          onClick={[Function]}
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "2.4%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="#"
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
</div>
`;
